### PR TITLE
Options for mesh adapt

### PR DESCRIPTION
**Pull request by mortezah**
This pull request includes a couple of logic for setting mesh adapt options. In particular,

1. logic to compute `in->maximumIterations` based on the requested size field and current mesh size.
2. logic to invoke Pre, Post, or Mid balancer if the mesh imbalance becomes larger than `in->maximumImbalance` during mesh adapt.

### CODE REVIEW

**Comment by mortezah** [comment]
@cwsmith I have added this condition https://github.com/mortezah/core/blob/f9005bb6c0f158a5b66c38953d4bd4318f01a320/ma/maInput.cc#L139 to prevent setting more than one of these flags `in->shouldRunPreZoltan`, `in->shouldRunPreZoltan`, and `in->shouldRunPreParma` to true at the same time. However, it seems that some of the tests (for example `refine2x.cc` here https://github.com/SCOREC/core/blob/f5025752630b03dff41251cad1ac829d8c275a3f/test/refine2x.cc#L107) turns on both `PreZoltanRib` and `PreParma`. Any idea why both options are turned on in `refine2x`?

**Review by cwsmith — CHANGES_REQUESTED** [review]
Looks good.  I submitted a PR to your branch for a small cmake change.

File: CMakeLists.txt [hunk 1]
```diff
@@ -120,6 +120,10 @@ if(ENABLE_OMEGA_H)
   bob_public_dep(Omega_h)
 endif()
 
+if(ENABLE_ZOLTAN)
+  add_definitions(-DHAVE_ZOLTAN)
```
[the remark this answers is not in the document]
Comment (mortezah) [reply CMakeLists.txt]: @cwsmith thanks. I pull those chagnes.

File: ma/maBalance.cc [hunk 2]
```diff
@@ -121,42 +121,145 @@ void printEntityImbalance(Mesh* m)
...
+  // Then, take care of the case where all the options are set to false.
+  // That is, if the default values have not changed by the user. In
+  // this case, we apply the best possible balancer, if weighted imbalance
+  // is bigger than in->maximumImbalance
+  if ((!in->shouldRunPreZoltan) &&
+      (!in->shouldRunPreZoltanRib) &&
+      (!in->shouldRunPreParma) &&
+      (estimateWeightedImbalance(a) > in->maximumImbalance)) {
+#ifdef HAVE_ZOLTAN
+    if (PCU_Comm_Peers() < 16000) {
```
[the remark this answers is not in the document]
Comment (mortezah) [reply ma/maBalance.cc]: @cwsmith I have added `MAX_ZOLTAN_GRAPH_RANK`. I have also updated the documentation in `maInput.h` for oxygen to reflect the behavior of mesh adapt based on these changes.

**Review by cwsmith — APPROVED** [review]

**Pull request merged**